src/main/java/simplifier/services/UserServiceImpl.java

```diff
@@ -26,12 +26,12 @@ public void setbCryptPasswordEncoder(
   }
 
   @Override
-  public User saveUser(User user) throws NameNotUniqueException{
+  public Optional<User> saveUser(User user) {
     User checkForUnique = userRepository.findByUsername(user.getUsername());
```

Comment:
Variable names should be noun but not verb. Change variable name to just user or foundUser